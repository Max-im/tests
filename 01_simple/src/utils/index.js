export const getNexId = arr => {
  let maxId = 0;
  arr.forEach(item => {
    if (maxId < item.id) maxId = item.id;
  });
  return [{ id: maxId + 1 }, ...arr];
};
