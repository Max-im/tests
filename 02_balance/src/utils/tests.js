import React from "react";
import renderer from "react-test-renderer";

export const isRender = component => {
  const c = renderer.create(component);
  expect(c.toJSON()).toMatchSnapshot();
};
