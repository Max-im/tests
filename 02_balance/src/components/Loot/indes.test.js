import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Loot } from "./";
import { isRender } from "../../utils/tests";
Enzyme.configure({ adapter: new Adapter() });

describe("Loot", () => {
  const mockFetch = jest.fn();
  const props = {
    balance: 10,
    bitcoin: { bpi: { USD: { rate: "1,000" } } },
    fetchBitcoin: mockFetch
  };
  const loot = shallow(<Loot {...props} />);

  test("render", () => isRender(loot));

  test("launch fetchBitcoin", () => expect(mockFetch).toHaveBeenCalled());

  test("display header correctly", () => {
    expect(loot.find("h3").text()).toBe("Bitcoin balance: 0.01");
  });
});
