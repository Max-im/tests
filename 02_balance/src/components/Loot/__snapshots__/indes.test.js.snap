// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot render 1`] = `
<div>
  <h3>
    Bitcoin balance: 
    0.01
  </h3>
</div>
`;
